### PR TITLE
fix: handle etherscan rate limit errors

### PR DESCRIPTION
**Pull request by matthewwalsh0**
**Description**

The mobile client uses the core `TransactionController` which supports the use of an API key when querying Etherscan for incoming or updated transactions.

The extension, using its `TransactionController`, only queries Etherscan for normal transactions using a single callout.

The mobile client, using the core `TransactionController`, queries Etherscan for both normal transactions and token transfer transactions related to an account. This requires 2 callouts and therefore consistently results in a rate limit error on one of the callouts if no API key is provided.

Previously, if either request failed due to a rate limit error, the update would be skipped, and no transactions would be added. This PR instead processes whatever transactions are found, regardless of which specific callout returned a rate limit error.

Since both callouts are performed simultaneously, it is currently random chance which callout results in the rate limit error, meaning both normal and token transfer transactions are still supported, but either could experience sporadic delays.

Additional changes include:

- Add back the background timer to ensure incoming transactions continue to be polled, even when the app is in background mode.
- Disable the `startBlock` argument in Etherscan requests to match the previous behaviour where a bug meant this was never being utilised.

**Recordings**

#### Transaction History After Import

https://github.com/MetaMask/metamask-mobile/assets/19745142/9867b984-cfc7-4225-8bf3-dd9cf640e27d

#### New Incoming Transaction & Notification

https://github.com/MetaMask/metamask-mobile/assets/19745142/b9a97442-b39a-4618-98b5-dd84a85698b2

**Issue**

Fixes [#1249](https://github.com/MetaMask/mobile-planning/issues/1249), [#1250](https://github.com/MetaMask/mobile-planning/issues/1250), [#1251](https://github.com/MetaMask/mobile-planning/issues/1251)

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

**Review by vinistevam — DISMISSED** [review]

**Review by OGPoyraz — DISMISSED** [review]

File: app/components/Nav/Main/index.js [hunk 1]
```diff
@@ -167,6 +169,10 @@ const Main = (props) => {
       // the background timer, which is less intense
       if (backgroundMode.current) {
         removeNotVisibleNotifications();
+
+        BackgroundTimer.runBackgroundTimer(async () => {
+          await TransactionController.updateIncomingTransactions();
+        }, AppConstants.TX_CHECK_BACKGROUND_FREQUENCY);
```
Comment (sethkfman) [inline app/components/Nav/Main/index.js:175]: This background activity may be flagged by the App Store reviewers and we may want to increase the timeout. Please annotate the background frequency on the description. Thx

Comment (matthewwalsh0) [reply app/components/Nav/Main/index.js:175]: Apologies for the confusion, this was mistakenly removed previously but matches the same timer and interval used in 7.6.

https://github.com/MetaMask/metamask-mobile/blob/v7.6.0/app/components/Nav/Main/index.js#L171

**Comment by sleepytanya** [comment]
All incoming transaction notifications work as expected, push notification are also seen when the app is on the background.
Will upload videos later.

**Review by sleepytanya — APPROVED** [review]
LGTM

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**